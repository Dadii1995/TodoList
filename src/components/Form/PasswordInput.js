import React from 'react'
import { FormGroup, Label, Input, FormFeedback } from 'reactstrap'

const PasswordInput = ({ field: { name, ...field }, form: { touched, errors }, ...props }) => {
  return (
    <FormGroup>
      {props.label && <Label>{props.label}</Label>}
      <Input
        autoComplete="off"
        data-cy={props.dataCy}
        type="password"
        {...field}
        aria-label={props.label}
        name={name}
      />
      {errors[name] && <FormFeedback>{errors[name]}</FormFeedback>}
    </FormGroup>
  )
}

export default PasswordInput
