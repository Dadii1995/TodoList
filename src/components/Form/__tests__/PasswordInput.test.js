import renderer from 'react-test-renderer'
import React from 'react'
import { Field, Formik } from 'formik'
import PasswordInput from '../PasswordInput'

const getWrapper = initialState => {
  const wrapper = renderer.create(
    <Formik errors={initialState.errors} initialValues={initialState.initialValues}>
      <Field component={PasswordInput} label={initialState.label} name={initialState.name} />
    </Formik>
  )
  const root = wrapper.root
  return { root, wrapper }
}
describe('<PasswordInput/>', () => {
  test('render password input with label', () => {
    const initialState = {
      name: 'passwordInput',
      label: 'Password',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render password input without label', () => {
    const initialState = {
      name: 'passwordInput',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render password input with initial value', () => {
    const initialState = {
      name: 'passwordInput',
      label: 'Password',
      initialValues: {
        passwordInput: 'Admin123',
      },
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render password input with error', () => {
    const initialState = {
      name: 'passwordInput',
      label: 'Password',
      errors: {
        passwordInput: 'Required',
      },
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
})
