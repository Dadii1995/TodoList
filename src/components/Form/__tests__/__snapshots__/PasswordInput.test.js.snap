// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput/> render password input with error 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Password
  </label>
  <input
    aria-label="Password"
    autoComplete="off"
    className="form-control"
    name="passwordInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
  />
</div>
`;

exports[`<PasswordInput/> render password input with initial value 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Password
  </label>
  <input
    aria-label="Password"
    autoComplete="off"
    className="form-control"
    name="passwordInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
    value="Admin123"
  />
</div>
`;

exports[`<PasswordInput/> render password input with label 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  >
    Password
  </label>
  <input
    aria-label="Password"
    autoComplete="off"
    className="form-control"
    name="passwordInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
  />
</div>
`;

exports[`<PasswordInput/> render password input without label 1`] = `
<div
  className="form-group"
>
  <input
    autoComplete="off"
    className="form-control"
    name="passwordInput"
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
  />
</div>
`;
